src/main/java/frc/robot/Vision.java

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.networktables.NetworkTableInstance;
+
+public class Vision {
+  private double angle = 0;
+  private double distance = 0;
+  private double targetInVision = 0;
+  private double[] defaultReturn = new double[6];
+
+  
+  /** Creates a new Vision. */
+  public Vision() {}
+
+  //calculate turning power
+  public double rotPower(double goalRotation)
+  {
+      double error = getAngle() - goalRotation;
+      double output = MathUtil.clamp(Constants.LimeLight.kP_rotate * error, -0.1, 0.1);
+      return output;
+  }
+
+  //valid target
+  public boolean validTarget(){
+    if(isTargetInVision()){
+      if(isTagInRange(getDistance())){
+        return true;
+      }
+    }
+    return false;
+  }
+
+  //calculate pivot angle
+  public double calculateGoalAngle()
+  {
+    return Constants.LimeLight.quadratic[0]*Math.pow(getDistance(),2) + Constants.LimeLight.quadratic[1]*getDistance() + Constants.LimeLight.quadratic[2];
+  }
+
+  //check if aprilTag is in shot range
+  public boolean isTagInRange(double distanceToTag){
+    return (distanceToTag >= Constants.LimeLight.shotRange[0] && distanceToTag <= Constants.LimeLight.shotRange[1]);
+  }
+
+  //get vision data atributes
+  public boolean isTargetInVision(){
+    fetchData();
+    return targetInVision > 0.9;
+  }
+
+  public double getAngle(){
+    fetchData();
+    return angle;
+  }
+
+  public double getDistance(){
+    fetchData();
+    return distance;
+  }
+
+  public double getTargetInVision(){
+    fetchData();
+    return targetInVision;
+  }
+
+  public void fetchData() {
+
+    //get pose of apriltag in camera space -> must configure in LL web GUI
+    targetInVision = NetworkTableInstance.getDefault().getTable("limelight").getEntry("tv").getDouble(0);
```

Comment:
lots of raw strings here, and if these are configurable by someone else, they should be constants in the vision class. Ex: if LL3 decides to change these values in the network tables and we want to reuse this code in the future, we may not understand what these values mean.